sphinxext/opengraph/visitor.py

```diff
@@ -0,0 +1,63 @@
+from docutils import nodes
+
+
+class OpenGraphVisitor(nodes.GenericNodeVisitor):
+    def __init__(
+        self, document: nodes.document, desc_len: int, title_count: int = 1
+    ) -> None:
+        super().__init__(document)
+        self.description = ""
+        self.desc_len = desc_len
+        self.title_count = title_count
+        self.first_list = ""  # todo: rename
+
+    def default_visit(self, node: nodes.Element) -> None:
```

Comment:
For Discord question 3, I think if statements are better here than `'visit_' + node_name` functions. So, as is currently.